content/en/blog/_posts/2023-12-19-taint-eviction-controller.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: blog
+title: "Kubernetes 1.29: Decouple taint-manager from node-lifecycle-controller"
+date: 2023-12-19
+slug: kubernetes-1-29-taint-eviction-controller
+---
+
+**Authors:** Yuan Chen (Apple), Andrea Tosatto (Apple)
+
+This blog discusses a new feature in Kubernetes 1.29 to improve the handling of taint-based pod eviction.
+
+## Background
+
+In Kubernetes 1.29, an improvement has been introduced to enhance the handling of taint-based pod eviction on nodes. This blog discusses the changes made to node-lifecycle-controller to separate its responsibilities and improve overall code maintainability.
+
+### Summary of changes
+
+node-lifecycle-controller previously combined two independent functions:
+
+- Adding a pre-defined set of `NoExecute` taints to Node based on the node's condition.
```

Comment:
In the previous comment, I suggested to change 'node' to 'Node' when you are referring to an API object. That is a generic style for our docs. I was expecting to check the page thoroughly for this situation, not just this single instance.

Reply:
> When you are generally discussing an API object, use [sentence-style capitalization](https://docs.microsoft.com/en-us/style-guide/text-formatting/using-type/use-sentence-style-capitalization).

&mdash; [style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-upper-camel-case-for-api-objects)

Writing “node” is OK, @tengqm (and @yuanchen8911) - especially in a blog article where the author's own style is allowed to show through.

Reply:
Writing "node" is okay sometimes, but it is still desirable to conform to a single style guide. When referencing to an API object, we use the object's kind name. For example, "Deployment" and "deployment", "Container" and "container" mean different things.

In line 20, "Adding ... to Node based on the node's condition" is not good.

Reply:
@tengqm , would you mind suggesting what it should be,  `Node` for both appearances ? Thanks.

Reply:
> In the previous comment, I suggested to change 'node' to 'Node' when you are referring to an API object. That is a generic style for our docs. I was expecting to check the page thoroughly for this situation, not just this single instance.

"Pod" and "PersistentVolume" are used. We prefer the lower case "pod" in "taint-based pod eviction".